src/lexer/comments.ts

```diff
@@ -64,18 +66,38 @@ export function skipSingleLineComment(
       break;
     }
     advanceChar(parser);
+    endLine = parser.line;
+    endColumn = parser.column;
     end++;
   }
-  if (parser.onComment)
+  if (parser.onComment) {
+    const loc = {
+      start: {
+        // FIXME: there is a bug for HTMLClose.
+        // the start loc of should be before \n-->
+        // which is end of last line.
```

Comment:
Note there is a bug for HTMClose comment, previous range can be easily patch with -4 (the length of `\n-->`), but loc is difficult because the column size of last line is unknown.
The result is the `loc.start` is something like `{line: 2, column: 0}` instead of `{line: 1, column: 14}`.

Reply:
not possible to use a local var to save the value before and after and subtract to get correct value? Why line change? Try local var

Reply:
The new line is consumed far before parsing first `-`.

https://github.com/meriyah/meriyah/blob/26d934b50ec018ccf7233ae3974be100178347ad/src/lexer/scan.ts#L351-L362

I am not sure how to save the column info when parsing new line.

https://github.com/meriyah/meriyah/blob/26d934b50ec018ccf7233ae3974be100178347ad/src/lexer/scan.ts#L545-L553

From what I read in ecma262, the starting of HTMClose comment should be before `\n-->`, that's where the line issue came.

Reply:
`parser.startLine` and `parser.startColumn` seem recorded last loc before newline.

Reply:
Try remove all instances of `parser.column` and use a `parser.columnIndex` etc. Then do get the `column` value, you can do
`parser.columnIndex - parser.index`. Ofc with `columnAfter` and `columnBefore`

Reply:
startLine and startColumn works, but I am running into another trouble.

For multiple comments like
```js
a;
// comment 1
// comment 2
/* comment 3 */
```
They are consumed by a single `scanSingleToken()` call, so the startLIne and startColumn are same for three comments.

I also could not update startLine and startColumn when parsing comments, because that will affect the `loc.end` on previous statement `a;` , because that statement is finalised after next token (all 3 comments) is scanned.

Reply:
what about using startIndex or what the name was? It updated on top of that func

Reply:
Wrote some not-pretty code to properly support the loc on HTMLClose comment.